src/hooks/useCurrentRoundId.ts

```diff
@@ -12,16 +14,23 @@ const contract = createVoidSignerVotingContractInstance(
 
 // This can be accessed without logging the user in.
 export default function useCurrentRoundId() {
+  const { addError } = useContext(ErrorContext);
+
   const { data, error, isFetching, refetch } = useQuery<string>(
     "roundId",
     () => {
-      return queryCurrentRoundId(contract).then((res) => {
-        if (res) {
-          return res;
-        } else {
+      return queryCurrentRoundId(contract)
+        .then((res) => {
+          if (res) {
+            return res;
+          } else {
+            return "";
+          }
+        })
+        .catch((err) => {
+          addError(err.message);
```

Comment:
would be great if `addError` could take the full error object and pull the message out in the context. this way you could cleanly call `.catch(addError)` in some cases (not necessarily in this case)

Reply:
I guess if we wanted to add custom messages (in some weird case where the error is nonsensical without context) I guess we could just throw a new error object we construct.